NuRadioReco/modules/voltageToEfieldConverter.py

```diff
@@ -31,65 +32,60 @@ def get_array_of_channels(station, use_channels, det, zenith, azimuth,
+
+    # take the intersection of all channels
+    t_min = np.max(t_min)
+    t_max = np.min(t_maxs)
+
+    n_samples = int((t_max - t_min) * channel.get_sampling_rate())
+    if n_samples % 2:
+        n_samples -= 1
+
+    window = NuRadioReco.framework.base_trace.BaseTrace()  # create base trace class to do the fft with correct normalization etc.
+    window.set_trace(np.zeros(n_samples), channel.get_sampling_rate(), t_min)
 
     traces = []
-    n_samples = None
     for iCh, channel in enumerate(station.iter_channels(use_channels)):
-        tstart = delta_t - (time_shifts[iCh] - tmin)
-        tstop = tmax - time_shifts[iCh] - delta_t + trace_length
-        iStart = int(round(tstart * channel.get_sampling_rate()))
-        iStop = int(round(tstop * channel.get_sampling_rate()))
-        if(n_samples is None):
-            n_samples = iStop - iStart
-            if(n_samples % 2):
-                n_samples -= 1
-
-        trace = copy.copy(channel.get_trace())  # copy to not modify data structure
-        trace = trace[iStart:(iStart + n_samples)]
-        if(debug_cut):
-            ax[iCh].plot(trace)
-        base_trace = NuRadioReco.framework.base_trace.BaseTrace()  # create base trace class to do the fft with correct normalization etc.
-        base_trace.set_trace(trace, channel.get_sampling_rate())
-        traces.append(base_trace)
+        tmp_channel = copy.copy(channel)  # copy to not modify data structure
+        if t_shifts[iCh]:
+            tmp_channel.apply_time_shift(t_shifts[iCh])  # apply time shift to channel
+
+        tmp_window = copy.copy(window)
+        tmp_window.add_to_trace(channel)
+        traces.append(tmp_window)
+
+
     times = traces[0].get_times()  # assumes that all channels have the same sampling rate
-    if(time_domain):  # save time domain traces first to avoid extra fft
+    if time_domain:  # save time domain traces first to avoid extra fft
         V_timedomain = np.zeros((len(use_channels), len(times)))
         for iCh, trace in enumerate(traces):
             V_timedomain[iCh] = trace.get_trace()
+
     frequencies = traces[0].get_frequencies()  # assumes that all channels have the same sampling rate
     V = np.zeros((len(use_channels), len(frequencies)), dtype=complex)
     for iCh, trace in enumerate(traces):
         V[iCh] = trace.get_frequency_spectrum()
 
-    efield_antenna_factor = signal_processing.get_efield_antenna_factor(station, frequencies, use_channels, det, zenith, azimuth, antenna_pattern_provider)
+    efield_antenna_factor = signal_processing.get_efield_antenna_factor(
+        station, frequencies, use_channels, det, zenith, azimuth, antenna_pattern_provider)
 
-    if(debug_cut):
-        plt.show()
+    if time_domain:
+        return times, efield_antenna_factor, V, V_timedomain
 
-    if(time_domain):
-        return efield_antenna_factor, V, V_timedomain
-
-    return efield_antenna_factor, V
+    return times, efield_antenna_factor, V
```

Comment:
Similar to comment above, modules that use this function will need to be updated for the different output format.

Reply:
I agree with Sjoerd, as this function can also be used stand-alone, this will be a cumbersome fix. It is not obvious why the code breaks. One needs to go into the code so see that the return values changed. 
I normally implemented this by allowing another kwarg `additional_output` which activate the return of additional output values.